cpp/log/etcd_consistent_store-inl.h

```diff
@@ -686,18 +670,21 @@ util::Status EtcdConsistentStore<Logged>::CleanupOldEntries() {
     }
     VLOG(1) << "Cleanup deleted unsequenced entry (" << util::ToBase64(hash)
             << ") corresponding to sequenced entry #" << sequence_number;
-
-    // Now delete the sequenced entry:
-    status = DeleteEntry(&entry);
-    if (!status.ok()) {
-      LOG(WARNING) << "Cleanup couldn't delete sequenced entry #"
-                   << sequence_number << " (but corresponding unsequenced "
-                   << "entry was successfully deleted) : " << status;
-      return status;
-    }
-    VLOG(1) << "Cleanup deleted sequenced entry #" << sequence_number;
   }
-  return util::Status::OK;
+  // Now remove mappings for deleted sequence numbers.
+  size_t to, from(i);
+  for (to = 0; from < sequence_mapping.Entry().mapping_size(); ++from, ++to) {
+    sequence_mapping.MutableEntry()->mutable_mapping()->SwapElements(from, to);
+    VLOG(1) << "Cleanup keeping sequence mapping for "
+            << sequence_mapping.Entry().mapping(to).sequence_number();
+  }
+  sequence_mapping.MutableEntry()->mutable_mapping()->DeleteSubrange(to, i);
+  for (auto m : sequence_mapping.Entry().mapping()) {
+    VLOG(1) << "Kept " << m.sequence_number() << " => "
+            << util::ToBase64(m.unsequenced_hash());
+  }
+  status = UpdateSequenceMapping(&sequence_mapping);
+  return status;
```

Comment:
I was going to say that you could ```return UpdateSequenceMapping(&sequence_mapping);``` directly, but I think a log message in case of an error would be nice?

It's possible that a lower level already does log errors, in which case, return straight up?